src/chatterbox_static_stream.erl

```diff
@@ -72,96 +74,91 @@ on_request_end_stream(State=#cb_static{connection_pid=ConnPid,
+                               ".ttf" -> <<"application/font-snft">>;
+                               _ -> <<"unknown">>
+                           end,
+                {ok, Data} = file:read_file(File),
+
+                ResponseHeaders = [
+                                   {<<":status">>, <<"200">>},
+                                   {<<"content-type">>, MimeType}
+                                  ],
+
+                case {MimeType, http2_connection:is_push(ConnPid)} of
+                    {<<"text/html">>, true} ->
+                        %% Search Data for resources to push
+                        {ok, RE} = re:compile("<link rel=\"stylesheet\" href=\"([^\"]*)|<script src=\"([^\"]*)|src: '([^']*)"),
+                        Resources = case re:run(Data, RE, [global, {capture,all,binary}]) of
+                                        {match, Matches} ->
+                                            [dot_hack(lists:last(M)) || M <- Matches];
+                                        _ -> []
+                                    end,
+                        lager:debug("Resources to push: ~p", [Resources]),
+
+                        NewStreams =
+                            lists:foldl(
+                              fun(R, Acc) ->
+                                      NewStreamId = http2_connection:new_stream(ConnPid),
+                                      PHeaders = generate_push_promise_headers(Headers, <<$/,R/binary>>),
+                                      http2_connection:send_promise(ConnPid, StreamId, NewStreamId, PHeaders),
+                                      [{NewStreamId, PHeaders}|Acc]
+                              end,
+                              [],
+                              Resources
+                             ),
+                        lager:debug("New Streams for promises: ~p", [NewStreams]),
+                        ok;
+                    _ ->
+                        ok
+                end,
+
+                %% For each chunk of data:
+
+                %% 1. Ask the connection if it's got enough bytes in the
+                %% send window.
+                %% maybe just send the frame header?
+
+                %% If it doesn't, we need to put this frame in a place
+                %% that will get looked at when our connection window size
+                %% increases.
+
+                %% If it does, we still need to try and check stream level
+                %% flow control.
+                {ResponseHeaders, Data};
         {false, false} ->
             ResponseHeaders = [
                                {<<":status">>,<<"404">>}
                               ],
-            http2_connection:send_headers(ConnPid, StreamId, ResponseHeaders),
-            http2_connection:send_body(ConnPid, StreamId, <<"No soup for you!">>),
-            ok
+                {ResponseHeaders, <<"No soup for you!">>}
+        end,
+
+    case {Method, HeadersToSend, BodyToSend} of
```

Comment:
which is what I'm doing, right? If the pattern matches <<"HEAD">> I just don't send a body